docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,85 @@
+# New delivery channel creation and integration
+
+## Status
+
+Proposed
+
+## Context
+
+Delivering notifications through different communication types (email, sms, web push notifications etc) is necessary in an extendable notifications service. Organizations should be able to choose which delivery channels (twilio, sendgrid, airship etc) they want to use for the associated communication type(s) as these said platforms could have different features or costs. Sometimes using different communication types through multiple delivery channels could save costs rather than having multiple communication types through a single delivery channel.
+
+## Decision
+```
+                 +------------+
+                 |            |
+                 | Management |
+                 | API        |
+                 |            |
+                 |            |
+                 ++--+------+-+
+                  ^  ^      ^
+                  |  |      |
+         +--------+  |      +--------------+
+         |           |                     |
+         v           v                     v
++--------+-+      +--+-------+           +-+--------+
+|          |      |          |           |          |
+| Delivery |      | Delivery |           | Delivery |
+| Channel  |      | Channel  | +-+  +-+  | Channel  |
+| 1        |      | 2        |           | N        |
+|          |      |          |           |          |
+|          |      |          |           |          |
++----------+      +----------+           +----------+
+```
+
+The management api should have knowledge of what delivery channels are available and where certain notification types should be sent. 
+
+When a new notification for a user is sent to the management api, the management api will read the user preference for the said notification type and see which communication types the user has 'subscribed' to. Following this, the notification delivered through the appropriate delivery channel.
+
+### User notifications and user settings
+
+A user should be able to choose what kind of communication types they receive based on the notification type and notification severity. There are two ways of approaching this, either:
```

Comment:
We don't have any concept of "notification severity" yet, so shouldn't talk about in this ADR. (If we want to introduce severity later, we can write a separate ADR which describes how it's expected to tie in with everything else).

Reply:
Agreed, I don't think it should be included in this ADR